### ISSUE TITLE
Documentation on how to configure CNI options

### ISSUE THREAD

**Issue by ccittadino-ctic** [rancher/rke#2679]
**RKE version: Whatever comes with Rancher 2.5.8**

**Docker version: 19.3.15**

**Operating system and kernel: Ubuntu Server 18 LTS**

**Type/provider of hosts: Azure**

**Cluster yaml:
```
answers: {}
docker_root_dir: /var/lib/docker
enable_cluster_alerting: false
enable_cluster_monitoring: false
enable_network_policy: false
fleet_workspace_name: fleet-default
local_cluster_auth_endpoint:
  enabled: true
name: REMOVED
rancher_kubernetes_engine_config:
  addon_job_timeout: 45
  authentication:
    strategy: x509|webhook
  authorization: {}
  bastion_host:
    ssh_agent_auth: false
  cloud_provider:
    azureCloudProvider:
      aad_client_cert_password: ''
      aad_client_id: REMOVED
      aad_client_secret: ''
      cloud_provider_backoff: false
      cloud_provider_backoff_duration: 0
      cloud_provider_backoff_exponent: 0
      cloud_provider_backoff_jitter: 0
      cloud_provider_backoff_retries: 0
      cloud_provider_rate_limit: false
      cloud_provider_rate_limit_bucket: 0
      cloud_provider_rate_limit_qps: 0
      location: REMOVED
      maximum_load_balancer_rule_count: 0
      resource_group: REMOVED
      subscription_id: REMOVED
      tenant_id: REMOVED
      use_instance_metadata: false
      use_managed_identity_extension: false
      vnet_resource_group: REMOVED
    name: azure
  dns:
    linear_autoscaler_params:
      cores_per_replica: 128
      max: 0
      min: 1
      nodes_per_replica: 4
      prevent_single_point_failure: true
    node_selector: null
    nodelocal:
      node_selector: null
      update_strategy:
        rolling_update: {}
    options: null
    reversecidrs: null
    stubdomains: null
    tolerations: null
    update_strategy:
      rolling_update: {}
    upstreamnameservers: null
  ignore_docker_version: true
  ingress:
    default_backend: false
    http_port: 0
    https_port: 0
    provider: nginx
  kubernetes_version: v1.20.9-rancher1-1
  monitoring:
    provider: metrics-server
    replicas: 1
  network:  # WHERE IS THE DOCUMENTATION ON HOW TO CONFIGURE THIS?
    canal_network_provider:
      iface: eth0
    mtu: 0
    options:
      canal_iface: eth0
      flannel_backend_type: vxlan
    plugin: canal
  restore:
    restore: false
  rotate_encryption_key: false
  services:
    etcd:
      backup_config:
        enabled: true
        interval_hours: 12
        retention: 6
        safe_timestamp: false
        timeout: 300
      creation: 12h
      extra_args:
        election-timeout: '5000'
        heartbeat-interval: '500'
      gid: 0
      retention: 72h
      snapshot: false
      uid: 0
    kube-api:
      always_pull_images: false
      pod_security_policy: false
      secrets_encryption_config:
        enabled: false
      service_node_port_range: 30000-32767
    kube-controller: {}
    kubelet:
      fail_swap_on: false
      generate_serving_certificate: false
    kubeproxy: {}
    scheduler: {}
  ssh_agent_auth: false
  upgrade_strategy:
    max_unavailable_controlplane: '1'
    max_unavailable_worker: 10%
    node_drain_input:
      delete_local_data: false
      force: false
      grace_period: -1
      ignore_daemon_sets: true
      timeout: 120
scheduled_cluster_scan:
  enabled: false
  scan_config:
    cis_scan_config:
      debug_master: false
      debug_worker: false
      override_benchmark_version: rke-cis-1.5
      profile: permissive
  schedule_config:
    cron_schedule: 0 0 * * *
    retention: 24
```
**

**Steps to Reproduce: **
1) Create a cluster with default options
2) Now try to figure out how to restrict the internal communication between nodes to private IPs
3) There are no detailed docs explaining how to do this. All that is available are some samples which do not work:
  a) https://rancher.com/docs/rke/latest/en/config-options/add-ons/network-plugins/#canal
  b) https://rancher.com/docs/rke/latest/en/config-options/nodes/#internal-address
  c) https://rancher.com/docs/rke/latest/en/example-yamls/

**Results:**
I've tried following the documentation to configure the default CNI (Canal) to use private ip's only by following the examples that are out there but they don't work (see config above).

**Comment by superseb** [comment]
The examples in the documentation refer to this being configured in the configmap: https://github.com/rancher/kontainer-driver-metadata/blob/dev-v2.5/rke/templates/canal.go#L13663. Please check if this is configured properly after you configured it according to the docs. Then please supply container logs where it should show what is being used/selected (or bump to a higher log level to see) and see if it gets applied.

Finally please provide the method and outcome you used to verify that the configured interface is not being used. (along with network configuration of the hosts involved)

**Comment by ccittadino-ctic** [comment]
Here is a screenshot of the interface I believe the traffic should be going over for private ip traffic:
![image](https://user-images.githubusercontent.com/71894697/131522071-941e8908-14a3-4980-a08f-316ded97ce97.png)

Here is the network config block from the cluster yaml in the Rancher UI:
```
  network:
    canal_network_provider:
      iface: eth0
    mtu: 0
    options:
      canal_iface: eth0
      flannel_backend_type: vxlan
```
Confirmed that the config map you referenced holds the same 'eth0' value:
```
apiVersion: v1
data:
  canal_iface: eth0
  cni_network_config: |-
    {
      "name": "k8s-pod-network",
      "cniVersion": "0.3.1",
      "plugins": [
        {
          "type": "calico",
          "log_level": "WARNING",
          "log_file_path": "/var/log/calico/cni/cni.log",
          "datastore_type": "kubernetes",
          "nodename": "__KUBERNETES_NODE_NAME__",
          "mtu": __CNI_MTU__,
          "ipam": {
              "type": "host-local",
              "subnet": "usePodCidr"
          },
          "policy": {
              "type": "k8s"
          },
          "kubernetes": {
              "kubeconfig": "__KUBECONFIG_FILEPATH__"
          }
        },
        {
          "type": "portmap",
          "snat": true,
          "capabilities": {"portMappings": true}
        },
        {
          "type": "bandwidth",
          "capabilities": {"bandwidth": true}
        }
      ]
    }
  masquerade: "true"
  net-conf.json: |
    {
      "Network": "10.42.0.0/16",
      "Backend": {
        "Type": "vxlan"
      }
    }
  typha_service_name: none
  veth_mtu: "1450"
kind: ConfigMap
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"v1","data":{"canal_iface":"eth0","cni_network_config":"{\n  \"name\": \"k8s-pod-network\",\n  \"cniVersion\": \"0.3.1\",\n  \"plugins\": [\n    {\n      \"type\": \"calico\",\n      \"log_level\": \"WARNING\",\n      \"log_file_path\": \"/var/log/calico/cni/cni.log\",\n      \"datastore_type\": \"kubernetes\",\n      \"nodename\": \"__KUBERNETES_NODE_NAME__\",\n      \"mtu\": __CNI_MTU__,\n      \"ipam\": {\n          \"type\": \"host-local\",\n          \"subnet\": \"usePodCidr\"\n      },\n      \"policy\": {\n          \"type\": \"k8s\"\n      },\n      \"kubernetes\": {\n          \"kubeconfig\": \"__KUBECONFIG_FILEPATH__\"\n      }\n    },\n    {\n      \"type\": \"portmap\",\n      \"snat\": true,\n      \"capabilities\": {\"portMappings\": true}\n    },\n    {\n      \"type\": \"bandwidth\",\n      \"capabilities\": {\"bandwidth\": true}\n    }\n  ]\n}","masquerade":"true","net-conf.json":"{\n  \"Network\": \"10.42.0.0/16\",\n  \"Backend\": {\n    \"Type\": \"vxlan\"\n  }\n}\n","typha_service_name":"none","veth_mtu":"1450"},"kind":"ConfigMap","metadata":{"annotations":{},"name":"canal-config","namespace":"kube-system"}}

    manager: kubectl-client-side-apply
    operation: Update
    time: "2021-08-27T15:38:19Z"
  name: canal-config
  namespace: kube-system
  resourceVersion: "439"
  uid: 7a170d9e-68e4-4b93-8afc-6df169f65fc2
```

What I know happens is that traffic still hits the Network Security Group (NSG in Azure) as a public ip, because I get this error when the cluster is provisioning:
```
[[network] Host [52.186.150.10] is not able to connect to the following ports: [20.102.54.102:6443, 52.186.150.10:6443]. Please check network policies and firewall rules]
```
Basically all etcd, internal k8s api, and monitoring traffic fails when I don't open up the firewall to allow all public traffic. If the interface were being honored then the traffic would come from the private ip space (192.x.x.x) correct?

Which logs would you like?

**Comment by superseb** [comment]
That error is not related to cluster network, that is a check between the configured host/node IP addresses. How are you registering your host/nodes? Are you specifying the IP addresses that need to be used (as specified on https://rancher.com/docs/rancher/v2.5/en/cluster-provisioning/rke-clusters/custom-nodes/agent-options/#ip-address-options) or are you using node drivers/templates?

**Comment by ccittadino-ctic** [comment]
The nodes are all node driven. In this case it uses an azure node template.

**Comment by ccittadino-ctic** [comment]
@superseb are you still able to assist us here? even though it's a node template?

**Comment by superseb** [comment]
I think you are running into https://github.com/rancher/rancher/issues/15944, but I have to check if that was added in the meantime.
